tutorials/knapsack_problem_with_displays.py

```diff
@@ -0,0 +1,187 @@
+        primitive1 = contour.plot_data(surface_style=surface_style)
+        text_style = TextStyle(text_color='rgb(0, 0, 0)',
+                               font_size=None,
+                               text_align_x='center',
+                               text_align_y='middle')
+        primitive2 = Text(comment=f'{self.mass} kg',
+                          position_x=0,
+                          position_y=0.4 + y_offset,
+                          text_style=text_style,
+                          text_scaling=True,
+                          max_width=0.5,
+                          multi_lines=False)
+        primitive3 = Text(comment=f'{self.price} €',
+                          position_x=0,
+                          position_y=-0.1 + y_offset,
+                          text_style=text_style,
+                          text_scaling=True,
+                          max_width=0.5,
+                          multi_lines=False)
+        return PrimitiveGroup(primitives=[primitive1, primitive2, primitive3])
+
+
+class Knapsack(PhysicalObject):
+    _standalone_in_db = True
+
+    def __init__(self, allowed_mass: float, name: str):
+        self.allowed_mass = allowed_mass
+        PhysicalObject.__init__(self, name=name)
+
+    def volmdlr_primitives(self):
+        height_vector = (self.allowed_mass + 0.5) * Z3D / 2
+        frame = Frame3D(origin=O3D + height_vector / 2,
+                        u=1.1 * X3D,
+                        v=1.1 * Y3D,
+                        w=height_vector + 0.1 * Z3D,
+                        name='frame ' + self.name)
+        primitives = [Block(frame=frame,
+                            alpha=0.3,
+                            name='block ' + self.name)]
+        return primitives
+
+
+class KnapsackPackage(Knapsack):
+    _vector_features = ["mass", "price", "golds", "silvers", "bronzes"]
+    _standalone_in_db = True
+
+    def __init__(self, items: List[Item], allowed_mass: float, name: str):
+        self.items = items
+        Knapsack.__init__(self, allowed_mass=allowed_mass, name=name)
+
+        self.mass = sum(item.mass for item in items)
+        self.price = sum(item.price for item in items)
+        self.golds = 0
+        self.silvers = 0
+        self.bronzes = 0
+        for item in items:
+            if item.color == 'gold':
+                self.golds += 1
+            elif item.color == 'silver':
+                self.silvers += 1
+            elif item.color == 'bronze':
+                self.bronzes += 1
+
+    def volmdlr_primitives(self):
+        primitives = super().volmdlr_primitives()
+        z_offset = 0
+        for item in self.items:
+            item_primitives = item.volmdlr_primitives(z_offset=z_offset)
+            primitives.extend(item_primitives)
+            z_offset += item.mass / 2 + 0.05
+        return primitives
+
+    @plot_data_view("2D display for KnapsackPackage")
+    def display_2d(self, reference_path: str = "#", **kwargs):
```

Comment:
Why add reference path here (2D view)?

For the new bundle we don't use reference_path anymore, for multiplotn (scatter and parallel) too !